### PR TITLE
this commit adds the GET /api/articles/:article_id endpoint

### PR DESCRIPTION
**Pull request by harrycs1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -33,4 +33,63 @@ describe('/api/topics', () => {
...
+describe('/api', () => {
+    test('GET: 200 sends an object describing all available endpoints to the user', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then(({ body }) => {
+            for (let endpoint in body) {
+                expect(body[endpoint]).toMatchObject({
+                    description: expect.any(String),
+                    queries: expect.any(Object),
```
Comment (ghost) [inline __tests__/app.test.js]: A tidy way to handle it, interesting.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -33,4 +33,63 @@ describe('/api/topics', () => {
...
+            }
+        })
+    })
+})
+
+describe('/api/articles', () => {
+    describe('api/articles/:article_id', () => {
+        test('GET:200 sends an article object to the user with the correct article_id', () => {
+            return request(app)
+            .get('/api/articles/2')
```
Comment (ghost) [inline __tests__/app.test.js:54]: You are specifying id 2, you specifically want article 2. Does your test give you any confidence that you have article 2 and its corresponding information?

File: controllers/topics.controllers.js [hunk 3]
```diff
@@ -9,4 +10,11 @@ exports.getTopics = (req, res, next) => {
     .catch((err) => {
         next(err);
     })
+}
+
+// this controller should be in it's own file?
```
Comment (ghost) [inline controllers/topics.controllers.js]: Don't leave comments in that don't mean anything. Yes controllers that have different jobs should be in different files. Yes you'll need to update it every time. And Yes you should really test that the new endpoint in the json file is there.

**Review by ghost — CHANGES_REQUESTED** [review]
Not a bad effort. However your test for your article needs to fundamentally be re written. You are being specific in your request, expecting any returned data is not sufficient. Please make those changes and resubmit.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -33,4 +34,55 @@ describe('/api/topics', () => {
...
+
+        test('GET:404 sends an appropriate status and error message when given a non-existent article_id', () => {
+            return request(app)
+              .get('/api/articles/88999')
+              .expect(404)
+              .then(({ body }) => {
+                expect(body.msg).toBe('Article does not exist');
+              });
+        });
+    })
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js:87]: nice sad path tests

**Review by nc-jonathan-mcguire — APPROVED** [review]
Looks all good to me 👍

**Pull request merged**